src/test/java/com/achdev/onlinebookstoreapp/controller/BookControllerTest.java

```diff
@@ -0,0 +1,547 @@
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.context.WebApplicationContext;
+
+@DisplayName("BookControllerTest Integration Test")
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class BookControllerTest {
+    protected static MockMvc mockMvc;
+    private static List<Book> books;
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Autowired
+    private BookRepository bookRepository;
+
+    @BeforeAll
+    static void setUp(
+            @Autowired DataSource dataSource,
+            @Autowired WebApplicationContext applicationContext
+    ) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+        setupDatabase(dataSource);
+        books = loadAllBooks();
+        scaleBookPrices(books);
+    }
+
+    @AfterAll
+    static void tearDown(@Autowired DataSource dataSource) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    private static void setupDatabase(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            executeSqlScripts(connection,
+                    "database/books/add-books-to-books-table.sql",
+                    "database/category/add-categories-to-categories-table.sql",
+                    "database/books/categories/add-books_categories-to_books_categories-table.sql"
+            );
+        }
+    }
+
+    @SneakyThrows
+    private static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            executeSqlScripts(connection,
+                    "database/books/categories/remove-all-from-book_categories-table.sql",
+                    "database/category/remove-categories-from-categories-table.sql",
+                    "database/books/remove-all-books-from-books-table.sql"
+            );
+        }
+    }
+
+    @WithMockUser(username = "user")
+    @DisplayName("Get all books")
+    @Order(1)
```

Comment:
why do you need to provide an order? your test cases should be isolated and work in any order